smdebug/core/config_validator.py

```diff
@@ -0,0 +1,85 @@
+# Standard Library
+import os
+from distutils.util import strtobool
+
+# First Party
+import smdebug.core.utils
+from smdebug.core.logger import get_logger
+from smdebug.core.utils import is_framework_version_supported
+from smdebug.profiler.profiler_config_parser import ProfilerConfigParser
+
+logger = get_logger()
+
+_config_validator = None
+
+
+class ConfigValidator(object):
+    def __init__(self, framework):
+        self._create_hook = strtobool(os.getenv("USE_SMDEBUG", "true").lower())
+        self._summary = ""
+        self._framework = framework
```

Comment:
Can we make the `framework` an enum? There are currently a lot of string checks based on the type of framework used, would be better to enforce an enum to reduce the chance of bugs that could come up.

Reply:
Done